### PR TITLE
Implements Default Fee

### PR DESCRIPTION
**Pull request by jacobburrell**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on small Android
- [ ] Tested on iOS Tablet
- [ x ] Tested on iOS Simulator
- [ ] n/a

### CODE REVIEW

File: src/components/scenes/ChangeMiningFeeScene.js [hunk 1]
```diff
@@ -196,21 +238,51 @@ const rawStyles = {
...
+      networkFeeOption: FeeOption,
+      isDefault: boolean,
+      currentDefault: DefaultFeeOption,
+      customFee: JsonObject,
+      walletId: string,
+      currencyCode?: string
+    ) {
+      console.log(`273: currentDefault: ${currentDefault}`)
+      console.log(`274: networkFeeOption: ${networkFeeOption}`)
+      console.log(`274: isDefault: ${isDefault.toString()}`)
```
Comment (thehobbit85) [inline src/components/scenes/ChangeMiningFeeScene.js:277]: remove

File: src/components/scenes/ChangeMiningFeeScene.js [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+import type { JsonObject } from 'edge-core-js'
+import type { EdgeCurrencyWallet, EdgeSpendTarget } from 'edge-core-js/types'
 import * as React from 'react'
 import { ScrollView, StyleSheet, Text, TouchableWithoutFeedback, View } from 'react-native'
 import { Actions } from 'react-native-router-flux'
 import EntypoIcon from 'react-native-vector-icons/Entypo'
 import { connect } from 'react-redux'
 
 import { sendConfirmationUpdateTx } from '../../actions/SendConfirmationActions.js'
+import { removeDefaultFeeSetting, setDefaultFeeSetting } from '../../actions/SettingsActions.js'
```
Comment (thehobbit85) [inline src/components/scenes/ChangeMiningFeeScene.js:12]: remove 'removeDefaultFeeSetting'

File: src/actions/SettingsActions.js [hunk 3]
```diff
@@ -129,6 +130,26 @@ export const setDenominationKeyRequest = (currencyCode: string, denominationKey:
...
+export const removeDefaultFeeSetting = (currencyCode: string) => (dispatch: Dispatch, getState: GetState) => {
+  // Thunk
+  const state = getState()
+  const { account } = state.core
+
+  return ACCOUNT_SETTINGS.removeDefaultFeeSetting(account, currencyCode)
+    .then(() => dispatch(SETTINGS_ACTIONS.removeDefaultFee(currencyCode)))
+    .catch(showError)
+}
+
```
Comment (thehobbit85) [inline src/actions/SettingsActions.js:152]: No need for this anymore

File: src/components/scenes/SendConfirmationScene.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ import { Slider } from '../../modules/UI/components/Slider/Slider.ui.js'
 import { type AuthType, getSpendInfoWithoutState } from '../../modules/UI/scenes/SendConfirmation/selectors'
 import { convertCurrencyFromExchangeRates } from '../../modules/UI/selectors.js'
 import { type GuiMakeSpendInfo, type SendConfirmationState } from '../../reducers/scenes/SendConfirmationReducer.js'
+import type { CurrencySetting } from '../../reducers/scenes/SettingsReducer'
```
Comment (thehobbit85) [inline src/components/scenes/SendConfirmationScene.js:26]: Why is this here?

File: src/components/scenes/SendConfirmationScene.js [hunk 5]
```diff
@@ -38,6 +39,7 @@ const DIVIDE_PRECISION = 18
 export type SendConfirmationStateProps = {
   fiatCurrencyCode: string,
   currencyInfo: EdgeCurrencyInfo | null,
+  currencySettings: CurrencySetting,
```
Comment (thehobbit85) [inline src/components/scenes/SendConfirmationScene.js:42]: Why is this here?

File: src/components/scenes/SendScene.js [hunk 6]
```diff
@@ -27,6 +27,7 @@ import { checkRecordSendFee, FIO_NO_BUNDLED_ERR_CODE } from '../../modules/FioAd
 import { Slider } from '../../modules/UI/components/Slider/Slider.ui'
 import { convertCurrencyFromExchangeRates } from '../../modules/UI/selectors.js'
 import { type GuiMakeSpendInfo } from '../../reducers/scenes/SendConfirmationReducer.js'
+import type { CurrencySetting } from '../../reducers/scenes/SettingsReducer.js'
```
Comment (thehobbit85) [inline src/components/scenes/SendScene.js:30]: Why is this here?

File: src/components/scenes/SendScene.js [hunk 7]
```diff
@@ -50,6 +51,7 @@ type StateProps = {
   authRequired: 'pin' | 'none',
   defaultSelectedWalletId: string,
   defaultSelectedWalletCurrencyCode: string,
+  currencySettings?: CurrencySetting,
```
Comment (thehobbit85) [inline src/components/scenes/SendScene.js:54]: Why is this here?

File: src/connectors/scenes/SendConfirmationConnector.js [hunk 8]
```diff
@@ -85,6 +85,7 @@ const mapStateToProps = (state: RootState): SendConfirmationStateProps => {
     balanceInCrypto,
     balanceInFiat,
     currencyCode,
+    currencySettings: state.ui.settings[getSelectedCurrencyCode(state)],
```
Comment (thehobbit85) [inline src/connectors/scenes/SendConfirmationConnector.js:88]: Why is this here?

File: src/modules/Core/Account/settings.js [hunk 9]
```diff
@@ -16,16 +17,14 @@ export const PASSWORD_RECOVERY_REMINDERS_SHOWN = {
...
-  autoLogoutTimeInSeconds: 3600,
-  defaultFiat: 'USD',
-  defaultIsoFiat: 'iso:USD',
-  merchantMode: false,
-  preferredSwapPluginId: '',
-  countryCode: '',
+type CurrencyDefault = {
+  [currency: string]: {
+    denomination?: string,
+    denomination?: string,
```
Comment (thehobbit85) [inline src/modules/Core/Account/settings.js:23]: I'm pretty sure one of them should be 'denominations', plural. 
How does it even let you commit this?

File: src/modules/Core/Account/settings.js [hunk 10]
```diff
@@ -667,6 +681,20 @@ export const setDenominationKeyRequest = (account: EdgeAccount, currencyCode: st
...
   })
+// Save default fee setting to disk
+export const saveDefaultFeeSetting = (account: EdgeAccount, currencyCode: string, defaultFee: DefaultFeeOption, customFee?: JsonObject) =>
+  getSyncedSettings(account).then(settings => {
+    const updatedSettings = updateCurrencySettings(settings, currencyCode, { defaultFee, customFee })
+    return setSyncedSettings(account, updatedSettings)
+  })
+
+// Remove default fee setting from disk
+export const removeDefaultFeeSetting = (account: EdgeAccount, currencyCode: string) =>
```
Comment (thehobbit85) [inline src/modules/Core/Account/settings.js:692]: No need for this anymore

File: src/modules/Settings/SettingsActions.js [hunk 11]
```diff
@@ -50,6 +53,18 @@ export const setDenominationKey = (currencyCode: string, denominationKey: string
...
 })
 
+// Set Default Fee
+export const setDefaultFee = (currencyCode: string, defaultFee: DefaultFeeOption, customFee: JsonObject) => ({
+  type: 'UI/SETTINGS/SET_DEFAULT_FEE',
+  data: { currencyCode: currencyCode, defaultFee: defaultFee, customFee: customFee }
+})
+
+// Remove Default Fee
+export const removeDefaultFee = (currencyCode: string) => ({
```
Comment (thehobbit85) [inline src/modules/Settings/SettingsActions.js:63]: No need for this anymore

File: src/reducers/scenes/SettingsReducer.js [hunk 12]
```diff
@@ -497,6 +504,20 @@ export const settingsLegacy = (state: SettingsState = initialState, action: Acti
...
       return { ...state, passwordRecoveryRemindersShown }
     }
+    case 'UI/SETTINGS/SET_DEFAULT_FEE': {
+      const { currencyCode, ...defaultFeeData } = action.data
+      return {
+        ...state,
+        [currencyCode]: { ...state[currencyCode], ...defaultFeeData }
+      }
+    }
+    case 'UI/SETTINGS/REMOVE_DEFAULT_FEE': {
```
Comment (thehobbit85) [inline src/reducers/scenes/SettingsReducer.js:514]: No need for this anymore

File: src/types/reduxActions.js [hunk 13]
```diff
@@ -191,6 +183,14 @@ export type Action =
         receiveAddress: EdgeReceiveAddress
       }
     }
+  | {
+      type: 'UI/SETTINGS/SET_DEFAULT_FEE',
+      data: { currencyCode: string, defaultFee: FeeOption, customFee: JsonObject }
+    }
+  | {
+      type: 'UI/SETTINGS/REMOVE_DEFAULT_FEE',
```
Comment (thehobbit85) [inline src/types/reduxActions.js:191]: No need for this anymore

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
Address all the comments and also add screen shots of the feature.

File: src/components/scenes/SendConfirmationScene.js (hunk 4 above)
Comment (jacobburrell) [reply src/components/scenes/SendConfirmationScene.js:26]: Originally it needed it when it was the main screen.

File: src/components/scenes/SendConfirmationScene.js (hunk 5 above)
Comment (jacobburrell) [reply src/components/scenes/SendConfirmationScene.js:42]: Originally it needed it when it was the main screen.

File: src/components/scenes/SendConfirmationScene.js (hunk 4 above)
Comment (jacobburrell) [reply src/components/scenes/SendConfirmationScene.js:26]: It's used in its state props:

```
import type { CurrencySetting } from '../../reducers/scenes/SettingsReducer'
import { THEME } from '../../theme/variables/airbitz.js'
import type { GuiCurrencyInfo, GuiDenomination, GuiWallet } from '../../types/types.js'
import { scale } from '../../util/scaling.js'
import { convertNativeToDisplay, convertNativeToExchange, decimalOrZero, getDenomFromIsoCode } from '../../util/utils.js'
import { AddressTextWithBlockExplorerModal } from '../common/AddressTextWithBlockExplorerModal'
import { ExchangeRate } from '../common/ExchangeRate.js'
import { SceneWrapper } from '../common/SceneWrapper.js'
import { ButtonsModal } from '../modals/ButtonsModal'
import { Airship, showError } from '../services/AirshipInstance'

const DIVIDE_PRECISION = 18

export type SendConfirmationStateProps = {
  fiatCurrencyCode: string,
  currencyInfo: EdgeCurrencyInfo | null,
  currencySettings: CurrencySetting,
```

**Comment by jacobburrell** [comment]
image
Screenshot requested

**Comment by jacobburrell** [comment]
image
Adjusted padding
